packages/wallets/src/features/cfd/flows/CTrader/AddedCTraderAccountsList/__tests__/AddedCTraderAccountsList.spec.tsx

```diff
@@ -1,62 +1,51 @@
-import React, { ReactNode } from 'react';
+import React from 'react';
 import { createMemoryHistory } from 'history';
 import { Router } from 'react-router-dom';
 import { useCtraderAccountsList } from '@deriv/api-v2';
 import { fireEvent, render, screen } from '@testing-library/react';
 import AddedCTraderAccountsList from '../AddedCTraderAccountsList';
-
-type TradingAccountCardProps = {
-    children: ReactNode;
-    leading: ReactNode;
-    trailing: ReactNode;
-};
+import { ModalProvider } from '../../../../../../components/ModalProvider';
 
 jest.mock('@deriv/api-v2', () => ({
     useCtraderAccountsList: jest.fn(),
 }));
 
-jest.mock('../../../../../../components/', () => ({
-    TradingAccountCard: ({ children, leading, trailing }: TradingAccountCardProps) => (
-        <div>
-            {leading}
-            {children}
-            {trailing}
-        </div>
-    ),
-}));
-
-const mockShow = jest.fn();
-jest.mock('../../../../../../components/ModalProvider', () => ({
-    useModal: () => ({ show: mockShow }),
-}));
-
-jest.mock('../../../../../../helpers/urls', () => ({
-    getStaticUrl: () => 'https://deriv.com/deriv-ctrader',
+jest.mock('../../../../modals', () => ({
+    MT5TradeModal: () => <div>Mocked MT5 Trade Modal</div>,
 }));
 
-jest.mock('../../../../../../public/images/ctrader.svg', () => {
-    const MockedSvg = () => <div>CTrader</div>;
-    MockedSvg.displayName = 'MockedSvg';
-    return MockedSvg;
-});
-
 describe('AddedCTraderAccountsList', () => {
+    let $root: HTMLDivElement, $modalContainer: HTMLDivElement;
     const history = createMemoryHistory();
     const mockAccounts = [
         { formatted_balance: '1000', login: '123' },
         { formatted_balance: '2000', login: '456' },
     ];
 
     beforeEach(() => {
+        jest.clearAllMocks();
+        $root = document.createElement('div');
```

Comment:
hmm is this really needed, is there any other way around? feels pretty weird to modify test containere from within the test itself

Reply:
cleaned this up and removed the root div but the div with the `wallets_modal_root` id is needed to mock and display the content of the modal, unsure if there are other ways to display the mocked modal content 🤔

Reply:
would that structure for rendering do the trick?
``` 
<div id="wallets_modal_root">
    <Router history={history}>
          <ModalProvider>
              <AddedCTraderAccountsList />
          </ModalProvider>
    </Router>
</div>
```

Reply:
or alternatively:
```
 <Router history={history}> 
     <div id="wallets_modal_root">
         <ModalProvider> 
             <AddedCTraderAccountsList />
           </ModalProvider> 
        </div> 
     </Router>
```
cc @lubega-deriv

Reply:
@wojciech-deriv these does not work and still fails the test, as we want a separate modal div to be rendered alongside the component div. This would wrap the component div in the modal div and not render the modal component itself
<img width="415" alt="image" src="https://github.com/binary-com/deriv-app/assets/142860499/7f7b6bef-256c-42b2-a586-2b40572a06dc">